contracts/programs/keystone-forwarder/src/state.rs

```diff
@@ -46,4 +46,7 @@ pub struct ExecutionState {
     pub transmission_id: [u8; 32],
     // until failure states are reported by the write target, success will always be true
     pub success: bool,
+    // having an explicit failure state allows us to differentiate between a failed transmission and one
+    // not yet initiated
+    pub failure: bool,
```

Comment:
Two options here:

1. 
You can tell if the transmission is being initialized for the first time because the transmission_id and transmitter will all be default 0 values. 

so for example, this will let you differentiate between what a "false" success means and one which was just initialized (so no report beforehand)

2. 
but to make it cleaner we can use enums so that 

enum Status {
    NotReported, // 0
    Success,     // 1
    Failure,     // 2
}

when you initialize the execution status for the first time it'll show NotReported.